examples/rcbd_two_var.R

```diff
@@ -0,0 +1,166 @@
+
+# Make sure the residuals are normal (this can also be seen in the Q-Q plot).
+cat('Shapiro-Wilk Normality Test\n')
+sep(50)
+#------------------------------------------------------------------------------#
+shapiro.test(residuals(model))
+#------------------------------------------------------------------------------#
+sep(50)
+
+# Run Levene's Test for a one-way ANOVA of each of the main factors.
+# TODO : clone is significant in Levene's test so transformation is necessary.
+cat("Levene's Test\n")
+sep(79)
+#------------------------------------------------------------------------------#
+leveneTest(yield ~ clone, data=my.data)
+leveneTest(yield ~ nitrogen, data=my.data)
+#------------------------------------------------------------------------------#
+sep(79)
+
+# Generate predicted values for Tukey 1-df Test
+# Perform a Tukey 1-df Test for Non-additivity
+cat("Tukey 1-df Test for Non-additivity\n")
+sep(79)
+#------------------------------------------------------------------------------#
+my.data$sq_preds <- predict(model)^2
+one.df.model <- lm(yield ~ clone + nitrogen + clone:nitrogen + block + sq_preds, my.data)
+anova(one.df.model)
+#------------------------------------------------------------------------------#
+sep(79)
+
+# Print the ANOVA table of the fit. The user will have to note the significant
+# factors. In this case all factors, including interaction nitrogen:clone, are
+# significant.
+cat('ANOVA Table\n')
+sep(50)
+#------------------------------------------------------------------------------#
+anova(model)
+#------------------------------------------------------------------------------#
+sep(50)
+
+# Show the confidence intervals.
+cat('Confidence Intervals\n')
+sep(50)
+#------------------------------------------------------------------------------#
+confint(model)
+#------------------------------------------------------------------------------#
+sep(50)
+
+# Plot the mean yield with respect to each clone for each N level and vice
+# versa.
+# TODO : This call to par() is not making the intxplots subplots for some
+# reason.
+dev.new()
+#------------------------------------------------------------------------------#
+par(mfrow = c(2, 1))
+intxplot(yield ~ clone, groups = nitrogen, data=my.data, se=TRUE,
+         ylim=range(my.data$yield), offset.scale=500)
+intxplot(yield ~ nitrogen, groups = clone, data=my.data, se=TRUE,
+         ylim=range(my.data$yield), offset.scale=500)
+#------------------------------------------------------------------------------#
+
+# The ANOVA table shows that each variable and the interaction are significant,
+# so we then see which levels of both clone and nitrogen are significant with
+# respect to each other using least significant difference.
+cat('Least Significant Difference\n')
+sep(50)
+#------------------------------------------------------------------------------#
+LSD.test(model, c('clone', 'nitrogen'), console=TRUE)
```

Comment:
If the interaction clone x nitrogen was NOT significant, we would want to run LSD.test() for each factor:
LSD.test(model, 'clone')
LSD.test(model, 'nitrogen')

This would also provide the 95% CI's for the LS means of the main treatment groups.